Proprification

juljac56 opened the pull request:
I took the previous reviews into account to render a cleaner code.
New file Student Meeting.css created as the css code was previously in the file BoxMeeting.css.  
Button "reset cropping" added in order to let the user cancels the cropping of the image.

juljac56 committed:
prise en compte reviews 1

juljac56 committed:
ajout bouton reset cadrage

juljac56 committed:
fix change room

juljac56 committed:
mofified readme

juljac56 committed:
remove dep window

juljac56 committed:
change first photo

djexvr commented on box-ui/README.md:
```diff
@@ -60,11 +60,15 @@ You must create a `.env` file (copy the `.env.template`), and fill it with the c
 Env variables starting with `REACT_APP_` are automatically catched by react, so we do not need the `dotenv` package.
 
-At the moment 3 variables are used :
+At the moment 6 variables are used :
```
Actually 7

djexvr commented on box-ui/src/components/ts/Student/StudentMeeting.tsx:
```diff
@@ -72,9 +72,11 @@ const StudentMeeting: FunctionComponent = () => {
-            const ws1 = new WebSocket(address);
+        const address_ws = process.env.REACT_APP_WS_ADDRESS;
+        if (address_ws == undefined) {
+            console.log('error websocket communication');
```
You can use console.error for error logs. If it is just for debug you should remove it.

djexvr commented on box-ui/src/components/ts/Student/StudentMeeting.tsx:
```diff
@@ -72,9 +72,11 @@ const StudentMeeting: FunctionComponent = () => {
-        const address = process.env.REACT_APP_WS_ADDRESS;
-        if (address) {
-            const ws1 = new WebSocket(address);
+        const address_ws = process.env.REACT_APP_WS_ADDRESS;
```
You should use camelCase as it is the convention for variables in the repo. (addressWs may be confusing so I recommend addressWS or addressWebsocket)
```suggestion
        const addressWebsocket = process.env.REACT_APP_WS_ADDRESS;
```

djexvr commented on box-ui/src/components/ts/Student/StudentMeeting.tsx:
```diff
@@ -26,16 +28,17 @@ const StudentMeeting: FunctionComponent = () => {
     const [textBtn, setTextBtn] = useState<string>('Recadrer');
 
     // img : downloaded from back, potentially cropped
     const [height_img, setHeight] = useState<number>(0);
```
You should use camelCase. The setter should also have the same name as the getter.
```suggestion
    const [heightImg, setHeightImg] = useState<number>(0);
```

djexvr commented on box-ui/src/components/ts/Student/StudentMeeting.tsx:
```diff
@@ -26,16 +28,17 @@ const StudentMeeting: FunctionComponent = () => {
 
     // img : downloaded from back, potentially cropped
     const [height_img, setHeight] = useState<number>(0);
+    const [width_img_double, setWidth_img_double] = useState<string>('');
```
```suggestion
    const [widthImgDouble, setWidthImgDouble] = useState<string>('');
```

djexvr commented on box-ui/src/components/ts/Student/StudentMeeting.tsx:
```diff
@@ -26,16 +28,17 @@ const StudentMeeting: FunctionComponent = () => {
+    const [imgIsCropped, setImgIsCropped] = useState<boolean>(false);
 
     // img original : not cropped
     const [width_img_original, setWidthOriginal] = useState<number>(0);
```
```suggestion
    const [widthImgOriginal, setWidthImgOriginal] = useState<number>(0);
```

djexvr commented on box-ui/src/components/ts/Student/StudentMeeting.tsx:
```diff
@@ -92,50 +94,53 @@ const StudentMeeting: FunctionComponent = () => {
-            const y = coord[i][1] * rapporty;
-            c = [...c, [x, y]];
+        coordinatesList = coord.map((point) => [point[0] * rapportx, point[1] * rapporty]);
+        const address_coord = process.env.REACT_APP_COORD;
```
```suggestion
        const addressCoord = process.env.REACT_APP_COORD;
```

djexvr commented on box-ui/README.md:
```diff
@@ -60,11 +60,15 @@ You must create a `.env` file (copy the `.env.template`), and fill it with the c
 -   `REACT_APP_MARSHA_URL` that set marsha's address
 -   `REACT_APP_POLICY_ADDRESS` that should be set to `[policy server | http://localhost:8070]/policy?custom_address=${roomName}` if you need to upload pictures on different servers (depending on the meeting for instance) otherwise you just need to go on /policy
 -   `REACT_APP_TIME_BETWEEN_PICTURES` defines the time between two pictures sent to backend server
+-   `REACT_APP_ORIGINAL_PHOTO` defines the address to get the image not processed
```
You don't have to define every route used in the .env, it would make it horrible to set up an instance if the app continues to grow. You should onlyset the Webroot (eg: https://back.jitsi-box.com) and then use it in the rest of the code 
```javascript
axios.get(`${process.env.BACK_WEBROOT}/websocket`)
```

djexvr replied:
Should be addressed in another branch as it is out of the scope of this one.

djexvr commented on box-ui/src/components/ts/Student/StudentMeeting.tsx:
```diff
@@ -92,50 +94,53 @@ const StudentMeeting: FunctionComponent = () => {
+        coordinatesList = coord.map((point) => [point[0] * rapportx, point[1] * rapporty]);
+        const address_coord = process.env.REACT_APP_COORD;
+        if (address_coord == undefined) {
+            console.log('error address coord');
```
Is it for debug if not the error message isn't really explicit

djexvr commented on box-ui/src/components/ts/Student/StudentMeeting.tsx:
```diff
@@ -92,50 +94,53 @@ const StudentMeeting: FunctionComponent = () => {
+        setImgIsCropped(false);
+        const address_coord = process.env.REACT_APP_COORD;
+        if (address_coord == undefined) {
+            console.log('error address coord');
```
The error message isn't really explicit

djexvr commented on box-ui/src/components/ts/Student/StudentMeeting.tsx:
```diff
@@ -92,50 +94,53 @@ const StudentMeeting: FunctionComponent = () => {
-            axios.post(address, { roomName: information.roomName, coord: c }).then(function () {
+    function resetCadrage() {
+        setImgIsCropped(false);
+        const address_coord = process.env.REACT_APP_COORD;
```
You should use camelCase

djexvr commented on box-ui/src/components/ts/Student/StudentMeeting.tsx:
```diff
@@ -178,26 +181,25 @@ const StudentMeeting: FunctionComponent = () => {
 
         newImg.onload = function () {
+            const width_new = newImg.width;
+            let height_new = newImg.height;
```
```suggestion
            const widthNew = newImg.width;
            let heightNew = newImg.height;
```

djexvr commented on box-ui/src/components/ts/Student/StudentMeeting.tsx:
```diff
@@ -178,26 +181,25 @@ const StudentMeeting: FunctionComponent = () => {
-                const height = newImg.height;
-                const width = newImg.width;
-                const ratio_img = ((width * 50) / height).toString() + 'vh';
+                const ratio_img = ((width_new * 45) / height_new).toString() + 'vh';
```
```suggestion
                const ratioImg = ((widthNew * 45) / heightNew).toString() + 'vh';
```

djexvr commented on box-ui/src/components/ts/Student/StudentMeeting.tsx:
```diff
@@ -178,26 +181,25 @@ const StudentMeeting: FunctionComponent = () => {
-                setWidthOriginal(width);
-                setHeightOriginal(height);
+                setWidthOriginal(width_new);
+                setHeightOriginal(height_new);
```
```suggestion
                setWidthOriginal(widthNew);
                setHeightOriginal(heightNew);
```

djexvr commented on box-ui/src/components/ts/Student/StudentMeeting.tsx:
```diff
@@ -178,26 +181,25 @@ const StudentMeeting: FunctionComponent = () => {
-                const pot_height = (height / width) * ViewportWidth;
+                const ViewportHeight = window.innerHeight * 0.45;
+                setWidth_img_double(((width_new * 45) / height_new).toString() + 'vh');
+                const pot_height = (height_new / width_new) * ViewportWidth;
```
```suggestion
                setWidthImgDouble(((width_new * 45) / heightNew).toString() + 'vh');
                const potHeight = (heightNew / widthNew) * ViewportWidth;
```

djexvr commented on box-ui/src/components/ts/Student/StudentMeeting.tsx:
```diff
@@ -178,26 +181,25 @@ const StudentMeeting: FunctionComponent = () => {
+                const pot_height = (height_new / width_new) * ViewportWidth;
                 if (pot_height < ViewportHeight) {
-                    height = pot_height;
+                    height_new = pot_height;
```
```suggestion
                    heightNew = potHeight;
```

djexvr commented on box-ui/src/components/ts/Student/StudentMeeting.tsx:
```diff
@@ -206,25 +208,27 @@ const StudentMeeting: FunctionComponent = () => {
-        const address = process.env.REACT_APP_PHOTO;
-        if (address) {
-            axios.get(address, { params: { roomName: information.roomName } }).then((resp) => {
+        const address_photo = process.env.REACT_APP_PHOTO;
```
```suggestion
        const addressPhoto = process.env.REACT_APP_PHOTO;
```

djexvr commented on box-ui/src/components/ts/Student/StudentMeeting.tsx:
```diff
@@ -290,23 +295,29 @@ const StudentMeeting: FunctionComponent = () => {
                 <div className='buttonAmeliorerVue'>
                     <button className='buttonStudent' onClick={() => AmeliorerVue()}>
-                        {textBtn}
+                        {textBtn == 'Recadrer' ? t('crop') : t('cancel')}
```
You should use english in hardcoded value. You  could also use a boolean.

djexvr committed:
📝  (readme) fix env variables count

djexvr committed:
🎨  (StudentMeeting) replace snake_case by camelCase

CamelCase is the convention in react and in this repo for variables.

djexvr committed:
🔊  (StudentMeeting) add explicit log on missing address

These logs should allow people hosting an instance to identify missing
env variables.

djexvr committed:
♻️  (StudentMeeting) remove useless state

The state `textBtn`is redundant and can be avoided.
THis makes the code lighter and is better for performances.

faydef committed:
Demo client (#69)

* implement MVP

* implements Suggestions

* ready for merge

Co-authored-by: Vianney Ruhlmann <85412254+djexvr@users.noreply.github.com>

djexvr committed:
Merge branch 'pre_prod' into proprification
